restapi/src/main/java/io/stargate/web/resources/Converters.java

```diff
@@ -489,63 +512,105 @@ public static Object toCqlValue(Column.ColumnType type, String value) {
-            return Float.parseFloat(value);
-          case Double:
-            return Double.parseDouble(value);
-          case Varint:
-            return new BigInteger(value);
-          case Decimal:
-            return new BigDecimal(value);
-          case Uuid:
-          case Timeuuid:
-            return UUID.fromString(value);
-          case Blob:
-            return ByteBuffer.wrap(Base64.getDecoder().decode(value));
-          case Inet:
-            return InetAddress.getByName(value);
-          case Date:
-            return LocalDate.parse(value);
-          case Time:
-            return LocalTime.parse(value);
-          case Timestamp:
-            return Instant.parse(value);
-          case Duration:
-            return CqlDuration.from(value);
-          default:
-            throw new AssertionError("Unsupported data type: " + rawType);
-        }
-      } catch (Exception e) {
+    }
+
+    if (rawType == Column.Type.Text
+        || rawType == Column.Type.Varchar
+        || rawType == Column.Type.Ascii) {
+      return value;
+    }
+
+    if (rawType == Column.Type.Boolean) {
+      // Boolean.parseBoolean returns false for anything not equal to "true"
+      if (value.equalsIgnoreCase("true")) {
+        return true;
+      } else if (value.equalsIgnoreCase("false")) {
+        return false;
+      } else {
         throw new IllegalArgumentException(
-            String.format("Invalid %s value: %s", rawType, e.getMessage()));
+            String.format("Invalid Boolean value '%s': cannot parse", value));
       }
     }
+
+    try {
+      if (rawType == Column.Type.Uuid || rawType == Column.Type.Timeuuid) {
+        return UUID.fromString(value);
+      }
+      if (rawType == Column.Type.Blob) {
+        return ByteBuffer.wrap(Base64.getDecoder().decode(value));
+      }
+      if (rawType == Column.Type.Duration) {
+        return CqlDuration.from(value);
+      }
+      if (rawType == Column.Type.Tinyint) {
+        return Byte.valueOf(value);
+      }
+      if (rawType == Column.Type.Smallint) {
+        return Short.valueOf(value);
+      }
+      if (rawType == Column.Type.Int) {
+        return Integer.valueOf(value);
+      }
+      if (rawType == Column.Type.Bigint || rawType == Column.Type.Counter) {
+        return Long.valueOf(value);
+      }
+      if (rawType == Column.Type.Float) {
+        return Float.parseFloat(value);
+      }
+      if (rawType == Column.Type.Double) {
+        return Double.parseDouble(value);
+      }
+      if (rawType == Column.Type.Varint) {
+        return new BigInteger(value);
+      }
+      if (rawType == Column.Type.Decimal) {
+        return new BigDecimal(value);
+      }
+    } catch (IllegalArgumentException e) {
+      // The exception's message doesn't really add information in any of those cases, use a generic
+      // description:
+      throw new IllegalArgumentException(
+          String.format("Invalid %s value '%s': cannot parse", rawType, value));
```

Comment:
In the original PR I avoided reflecting back the value because I was worried that would open the door to XSS attacks. But actually this is safe, we wrap the error in a JSON response in `RequestHandler`, so the message will be properly sanitized.